pkg/core/serving/common/common.go

```diff
@@ -60,49 +69,49 @@ func GenOpenFunctionContext(s *openfunction.Serving, components map[string]*comp
 		version = *s.Spec.Version
 	}
 
-	fc := functioncontext.Context{
+	fc := functionContext{
 		Name:    functionName,
 		Version: version,
-		Runtime: functioncontext.Runtime(s.Spec.Runtime),
+		Runtime: string(s.Spec.Runtime),
 		Port:    fmt.Sprintf("%d", port),
 	}
 
 	switch s.Spec.Runtime {
 	case openfunction.Async:
 		if s.Spec.Inputs != nil && len(s.Spec.Inputs) > 0 {
-			fc.Inputs = make(map[string]*functioncontext.Input)
+			fc.Inputs = make(map[string]*functionInput)
 
 			for _, i := range s.Spec.Inputs {
```

Comment:
i needs to be deepcopy to another local variable in loop to use i.Name or i.Param later